### PR TITLE
Add analytics to content safety events

### LINKED ISSUE #2921: Add analytics events for content safety interactions
## Description

The content safety project describes the following [analytics events](https://docs.openverse.org/projects/proposals/trust_and_safety/detecting_sensitive_textual_content/20230309-project_proposal_detecting_sensitive_textual_content.html#id3).

With some slight tweaks, these are the events.

- [ ] <s>`TOGGLE_SENSITIVE_RESULTS`</s> `TOGGLE_FETCH_SENSITIVE`
- [ ] <s>`TOGGLE_DO_NOT_BLUR`</s> `TOGGLE_BLUR_SENSITIVE`
- [ ] `UNBLUR_SENSITIVE_RESULT`
- [ ] `REBLUR_SENSITIVE_RESULT`
- [ ] `GO_BACK_FROM_SENSITIVE_RESULT`

The goal is to define them and send them to Plausible from appropriate places in the frontend.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #2921 by @dhruvkb

## Description

This PR adds 5 analytics events and calls them from appropriate places in the frontend.

### Prep

1. Check out the PR and run the dev server.
1. Open the browser console to keep watch over the analytics events.
1. Enable the `sensitive_content` feature flag from `/preferences`.
   You can also enable `fake_sensitive` if you'd rather not see actual sensitive content.

### Toggle events

1. Switch on the "Sensitive results" toggle.
   Look for the `TOGGLE_FETCH_SENSITIVE {checked: true}` message.
1. Switch off the "Blur content" toggle.
   Look for the `TOGGLE_BLUR_SENSITIVE {checked: false}` message.
1. Switch off the "Sensitive results" toggle.
   Look for the `TOGGLE_FETCH_SENSITIVE {checked: false}` and `TOGGLE_BLUR_SENSITIVE {checked: true}` messages.

### Wall interaction events

1. Pick a sensitive result and click on it.
   Look for the `SELECT_SEARCH_RESULT` message (that's not new).
1. Click on the "Back to results" button.
   Look for the `GO_BACK_FROM_SENSITIVE_RESULT {id: '...', sensitivities: '...'}` message.
1. Pick a sensitive result and click on it.
1. Click on the "Show content" button.
   Look for the `UNBLUR_SENSITIVE_RESULT {id: '...', sensitivities: '...'}` message.
1. Click on the "Hide content" button above the result.
   Look for the `REBLUR_SENSITIVE_RESULT {id: '...', sensitivities: '...'}` message.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — APPROVED** [review]
I tested all the events and they all work well.

Do you think it would make sense to add a "select_sensitive_result" event? Or, maybe, add "isSensitive" and "isBlurred" to the existing "select_result" event? This way we could track how many sensitive results the users click.

File: frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue [hunk 1]
```diff
@@ -65,12 +66,14 @@ export default defineComponent({
...
 
     const featureFlagStore = useFeatureFlagStore()
+    const { sendCustomEvent } = useAnalytics()
 
     let fetchSensitive = computed(() =>
       featureFlagStore.isOn("fetch_sensitive")
     )
     let setFetchSensitive = ({ checked }: { checked: boolean }) => {
       featureFlagStore.toggleFeature("fetch_sensitive", checked ? ON : OFF)
+      sendCustomEvent("TOGGLE_FETCH_SENSITIVE", { checked })
```
Comment (obulat) [inline frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue:76]: `checked` can be `undefined` as well. This change was added in #2897.

Comment (dhruvkb) [reply frontend/src/components/VSafeBrowsing/VSafeBrowsing.vue:76]: I don't know what needs to be changed here. Type check is passing.

**Review by AetherUnbound — APPROVED** [review]
All the analytics events worked during my local testing, and the code looks good!

**Comment by dhruvkb** [comment]
Created #2969 to include @obulat's review suggestion.

**Pull request merged**